tutorials/v7/climate/climate_tutorial.C

```diff
@@ -0,0 +1,249 @@
+   auto dfSummer = df.Filter(fnSummer, {"Month"});
+   cout << "The count for Summer: " << *dfSummer.Count() << std::endl;
+   auto dfFall = df.Filter(fnFall, {"Month"});
+   cout << "The count for Fall: " << *dfFall.Count() << std::endl;
+
+   // Create an orange histogram for fall
+   auto fallHist = dfFall.Histo1D({"Fall Average Temp", "Average Temperature by Season", 100, -40, 40}, "AverageTemperature");
+   fallHist->SetLineColor(kOrange);
+   fallHist->SetLineWidth(6);
+   // Create a blue histogram for winter
+   auto winterHist = dfWinter.Histo1D({"Winter Average Temp", "Average Temperature by Season", 100, -40, 40}, "AverageTemperature");
+   winterHist->SetLineColor(kBlue);
+   winterHist->SetLineWidth(6);
+   // Create a green histogram for spring
+   auto springHist = dfSpring.Histo1D({"Spring Average Temp", "Average Temperature by Season", 100, -40, 40}, "AverageTemperature");
+   springHist->SetLineColor(kGreen);
+   springHist->SetLineWidth(6);
+   // Create a red histogram for summer
+   auto summerHist = dfSummer.Histo1D({"Summer Average Temp", "Average Temperature by Season", 100, -40, 40}, "AverageTemperature");
+   summerHist->SetLineColor(kRed);
+   summerHist->SetLineWidth(6);
+
+   // Functions to filter for the time period between 2003-2013, and 1993-2002
+   auto fn1993To2002 = [](int year) { return year >= 1993 && year <= 2002; };
+   auto fn2003To2013 = [](int year) { return year >= 2003 && year <= 2013; };
+
+   // Create a RDataFrame for decades 1993_to_2002 & 2003_to_2013 by filtering ntupleToDF
+   // Display the count for each decade
+   auto df1993To2002 = df.Filter(fn1993To2002, {"Year"});
+   cout << "\nThe count for 1993_to_2002: " << *df1993To2002.Count() << std::endl;
+   auto df2003To2013 = df.Filter(fn2003To2013, {"Year"});
+   cout << "The count for 2003_to_2013: " << *df2003To2013.Count() << std::endl;
+
+   // Create a violet histogram for 1993_to_2002
+   auto hist_1993_to_2002 = df1993To2002.Histo1D({"1993_to_2002 Average Temp", "Average Temperature: 1993_to_2002 vs. 2003_to_2013", 100, -40, 40}, "AverageTemperature");
+   hist_1993_to_2002->SetLineColor(kViolet);
+   hist_1993_to_2002->SetLineWidth(6);
+   // Create a spring-green histogram for 2003_to_2013
+   auto hist_2003_to_2013 = df2003To2013.Histo1D({"2003_to_2013 Average Temp", "Average Temperature: 1993_to_2002 vs. 2003_to_2013", 100, -40, 40}, "AverageTemperature");
+   hist_2003_to_2013->SetLineColor(kSpring);
+   hist_2003_to_2013->SetLineWidth(6);
+
+   // Create a canvas to display histograms for average temperature by season.
+   auto canvas = RCanvas::Create("Average Temperature by Season");
+   canvas->Draw<TObjectDrawable>(std::shared_ptr<TH1D>(fallHist.GetPtr()), "L");
+   canvas->Draw<TObjectDrawable>(std::shared_ptr<TH1D>(winterHist.GetPtr()), "L");
+   canvas->Draw<TObjectDrawable>(std::shared_ptr<TH1D>(springHist.GetPtr()), "L");
+   canvas->Draw<TObjectDrawable>(std::shared_ptr<TH1D>(summerHist.GetPtr()), "L");
```

Comment:
This does not work properly. `RResultPtr<TH1D>` instance like `fallHist` keep ownership over the histogram object.
It is destroyed immediately when leaving function scope. Therefore nothing in disaplayed in  the `RCanvas` without explicit `RCanvas::Update` call. The only way to solve problem - create copy of all histograms and only then use them in RCanvas. Something like:
```
   auto h1 = (TH1D *) fallHist.GetPtr()->Clone(); h1->SetDirectory(nullptr);
   auto h2 = (TH1D *) winterHist.GetPtr()->Clone(); h2->SetDirectory(nullptr);
   auto h3 = (TH1D *) springHist.GetPtr()->Clone(); h3->SetDirectory(nullptr);
   auto h4 = (TH1D *) summerHist.GetPtr()->Clone(); h4->SetDirectory(nullptr);

   // Create a canvas to display histograms for average temperature by season.
   auto canvas = RCanvas::Create("Average Temperature by Season");
   canvas->Draw<TObjectDrawable>(std::shared_ptr<TH1D>(h1), "L");
   canvas->Draw<TObjectDrawable>(std::shared_ptr<TH1D>(h2), "L");
   canvas->Draw<TObjectDrawable>(std::shared_ptr<TH1D>(h3), "L");
   canvas->Draw<TObjectDrawable>(std::shared_ptr<TH1D>(h4), "L");
   // Create a legend for the seasons canvas
   auto legend = std::make_shared<TLegend>(0.1,0.7,0.48,0.9);
   legend->AddEntry(h1,"fall","l");
   legend->AddEntry(h2,"winter","l");
   legend->AddEntry(h3,"spring","l");
   legend->AddEntry(h4,"summer","l");
   canvas->Draw<TObjectDrawable>(legend, "L");
   canvas->Show();
```
Same should be applied for the second canvas.
Then all extra `Update` calls can be removed

Reply:
Many thanks for looking into it! @JohnYoon13 Let's go for this approach.  I suggest to make a helper function the provides a shared pointer of the input histogram  Something like

```
std::shared_ptr<TH1D> GetDrawableHist(const RResultPtr<TH1D> &h) {
   auto result = std::shared_ptr<TH1D>(static_cast<TH1D *>(h.GetPtr()->Clone()));
   result->SetDirectory(nullptr);
   return result;
}
```

Let's flag that helper with a `//TODO` to remind ourselves that we'd like to solve this in a better way in the future.